src/main/java/com/pinterest/secor/parser/TimestampedMessageParser.java

```diff
@@ -49,12 +72,101 @@ protected static long toMillis(final long timestamp) {
         return timestampMillis;
     }
 
+    protected String[] generatePartitions(long timestampMillis, boolean usingHourly)
```

Comment:
In the hourly partitioner, we are generating both hourly and daily partition string.  So the caller of this method needs to decide whether for a particular partition string, whether a daily format or a hourly format should be used (even though mUsingHourly is always true for that partitioner).